webapp/src/components/github_repo_selector/github_repo_selector.jsx

```diff
@@ -34,7 +34,16 @@ export default class GithubRepoSelector extends PureComponent {
     }
 
     onChange = (name, newValue) => {
-        this.props.onChange(newValue);
+        const newID = this.props.yourRepos.reduce((id, repo) => {
+            return repo.full_name === newValue ? repo.id : id;
+        }, 0);
+
+        // return if there's no match (ideally this should never happen)
+        if (newID === 0) {
+            return;
+        }
+
+        this.props.onChange(newID, newValue);
```

Comment:
We retrieve labels by repository ID, so we have to save it here along with the name.